quarkus/config-api/src/main/java/org/keycloak/config/EventOptions.java

```diff
@@ -23,8 +23,24 @@ public class EventOptions {
             .category(OptionCategory.EVENTS)
             .description("Comma-separated list of events to be collected for user event metrics. Reduce the number of metrics. If empty or not set, all events create a metric.")
             .buildTime(false)
+            .expectedValues(sortedListOfEvents())
+            .deprecatedMetadata(DeprecatedMetadata.deprecateValues("Use `remove_credential` instead of `remove_totp`, and `update_credential` instead of `update_totp` and `update_password`.", "remove_totp", "update_totp", "update_password"))
             .build();
 
+    private static List<String> sortedListOfEvents() {
+        List<String> events = new java.util.ArrayList<>(List.of("register", "login", "code_to_token", "logout", "client_login", "refresh_token", "introspect_token", "federated_identity_link",
```

Comment:
how do we maintain this list in future, if something changes as there are references of what are the possible event types in other places of the project. Would it better to centralize this list and modify it from one place ?

Reply:
The unit test will fail of we add or remove an entry from the list in another place of the project and miss to update this list. So it will always be consistent. 

We can have a follow-up issue to discuss with the the CND team how to make the enum accessible to the CLI module.